pkg/api/util/warnings.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+// Warning values thrown by validating webhook
+// https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/
+const (
+	// DeprecatedACMEEABKeyAlgorithmField is raised when the deprecated keyAlgorithm field for an ACME issuer's external account binding (EAB) is set.
+	DeprecatedACMEEABKeyAlgorithmField = "ACME issuer spec field 'externalAccount.keyAlgorithm' is deprecated. The value of this field will be ignored."
```

Comment:
This const isn't referenced outside of the `pkg/internal/apis/certmanager/validation` package. I think it should live in there as an un-exported const (or alternatively just hardcoded as the message in the call-site, which has the side effect of forcing us to write it out explicitly in test cases too, so we know if the message has been changed unintentionally)

Reply:
I thought that in this case there is no harm in exporting this const and having all the future warnings in the same place would help us with consistency of the wording and naming + make it easier to change the warning text (i.e if we later want to add a version at which this field would be removed). 
If anyone was actually parsing these warnings, I'd rather them use the const, then hardcode strings.

Reply:
My concern is that if we start treating this as an API surface, it makes minor re-wording/edits harder in future (e.g. if we ever want to change this text).

I don't think we'd want to encourage people to parse this text, and it seems like it is harder for us to read the code that's actually emitting these warnings if there's a cross-package reference.

If this _is_ to be exported and parseable, what's the versioning story here?

Reply:
If this were more akin to the `reason` field on a condition (i.e. short, all one word, etc) then I'd definitely agree that there's value in having them exported. This seems more akin to the 'message' field on a condition, which is explicitly designed to be human readable.

As far as I'm aware, k8s itself doesn't treat warning messages as part of its API surface and does not advise building automation on top of parsing these strings

Reply:
Good point about versioning, I didn't think about that.

I guess because these warnings would be returned to whoever applies the yaml (rather than just apearing in the logs), I thought it'd be nice to have them consistently worded, which might be easier achievable if they are all in the same place vs a bunch of hardcoded strings across the codebase.

Reply:
👍 re consistency - what do you think about just moving them to where they are actually used (unexported to convey they shouldn't be depended upon/parsed), but keeping them as consts for consistencies sake? I just think being in an 'api/' package insinuates it forms part of our API (and thus can be depended upon in a similar manner to other parts of our API)

Reply:
Thanks for the comments!
I've made the warning value unexported.
I did still put it into a file of its own in the internal validation package because I felt like this might help with the consistency of any future validation warnings that we might want to add- happy to be convinced otherwise.